assets/production-bestpractices.md

```diff
@@ -0,0 +1,89 @@
+
+1. Use an external etcd cluster
+2. Run a compute only StorageOS daemonset where the etcd masters will run based on node labels. 
+
+## External etcd cluster (+v3.0).
+
+Deploy an external etcd cluster in high availability. Either 3 or 5 nodes to keep quorum.
+
+You can find an example of the manifests in this [repository](https://github.com/storageos/demos/tree/master/k8s/deploy-storageos/external-etcd):
+
+```
+git clone https://github.com/storageos/demos.git storageos-examples
+cd storageos-examples/demos/k8s/deploy-storageos/external-etcd
+```
+
+Once the etcd cluster is up and running, connect StorageOS to this cluster. To do so you can define a Kubernetes Service and Endpoints to create an interface with the external etcd nodes. Edit the `manifests/025_etcd_service.yaml`, so the Endpoint' subsets.addresses point to your etcd addresses:
+
+```
+kind: Service
+apiVersion: v1
+metadata:
+  name: etcd-storageos
+  namespace: storageos
+spec:
+  ports:
+  - protocol: TCP
+    port: 2379
+---
+kind: Endpoints
+apiVersion: v1
+metadata:
+    name: etcd-storageos
+    namespace: storageos
+subsets:
+    - addresses: # IPs of your etcd servers
+        - ip: 172.28.128.14
+        - ip: 172.28.128.15
+        - ip: 172.28.128.3
+      ports:
+        - port: 2379
+```
+
+After that, edit the `manifests/040_daemonset.yaml_template`. The deployment script fills the `KV_ADDR` with the ip of the Kubernetes Service. If your Kubernetes cluster can resolve the service name `etcd-storageos`, you can define that value as follows:
+
+```
+        env:
+           - name: KV_BACKEND
+             value: 'etcd'
+           - name: KV_ADDR
+             value: 'etcd-storageos:2379'
+```
+
+## Compute Only DaemonSet
+
+Deploy a compute only StorageOS DaemonSet on the nodes that will be responsible for running etcd servers, either 3 or 5 nodes. Once these pods are up and running, deploy the full StorageOS DaemonSet joining the same cluster.
+   
+First, label the Kubernetes nodes that host the StorageOS compute only DaemonSet:
+
+```
+kubectl label node my_node1 my_node2 my_node3 storageos=compute-only
+```
+
+Then label the rest of the nodes that will run StorageOS:
+
+```
+kubectl label node my_nodeX storageos=storage
+```
+
+> Nodes not labeled won't run StorageOS in this installation procedure. Pods running on nodes where StorageOS is not present cannot mount volumes managed by StorageOS.
```

Comment:
"Nodes not labeled" => "Unlabelled nodes"

Reply:
done